solidity/contracts/utility/PriceOracle.sol

```diff
@@ -109,4 +149,13 @@ contract PriceOracle is IPriceOracle, Utils {
 
         return (numerator, denominator, lastUpdateTime());
     }
+
+    /** @dev returns the decimals of a given token */
+    function decimals(IERC20Token _token) private view returns (uint8) {

```

Comment:
Another option here is to implement this in assembly, and return 18 on failure.
This way, you cover both ETH and any non-standard ERC20 token.

Reply:
It's too risky since some tokens use different decimals, while not implementing the `decimals` interface. We can always change and upgrade it in the future.

Reply:
Yes, I know. For these tokens, your current method **will** fail.

Reply:
That's by design and as requested, though